lib/IRGen/IRGenSIL.cpp

```diff
@@ -2411,6 +2412,15 @@ void IRGenSILFunction::visitFullApplySite(FullApplySite site) {
                                    site.getSubstitutionMap(),
                                    &witnessMetadata, llArgs);
 
+  // Check if this call site has a callee marked
+  // with the musttail semantics attribute
+  // if it is then mark the emmision.
+  if (SILFunction *f = site.getReferencedFunctionOrNull()) {
+    if (f->hasSemanticsAttr(semantics::MUST_TAIL)) {
+      emission.setMustTail();
+    }
+  }
```

Comment:
I can't quite tell when this is null... Is that possible?

Reply:
Yes. It is if we can't find the source.